pages/dnd/reorderable-list.tsx

```diff
@@ -0,0 +1,105 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { ForwardedRef, forwardRef } from 'react';
+import { SyntheticListenerMap } from '@dnd-kit/core/dist/hooks/utilities';
+import { useUniqueId } from '@dnd-kit/utilities';
+import clsx from 'clsx';
+
+import { Box, SpaceBetween } from '~components';
+import { DndContainer } from '~components/internal/components/dnd-container';
+import DragHandle from '~components/internal/components/drag-handle';
+
+import { Instance } from '../table/generate-data';
+import { DnsName, i18nStrings, Status } from './commons';
+
+import styles from './styles.scss';
+
+interface OptionProps<Option> {
+  ref?: ForwardedRef<HTMLDivElement>;
+  option: Option;
+  dragHandleAriaLabel?: string;
+  listeners?: SyntheticListenerMap;
+}
+
+export function ReorderableList<Option extends { id: string }>({
+  options,
+  onChange,
+  renderOption,
+  renderStaticOption,
+  fixedOptionsStart = 0,
+}: {
+  options: readonly Option[];
+  onChange: (options: readonly Option[]) => void;
+  renderOption: (props: OptionProps<Option>) => React.ReactNode;
+  renderStaticOption: (option: Option) => React.ReactNode;
+  fixedOptionsStart?: number;
+}) {
+  const staticOptions = options.slice(0, fixedOptionsStart);
+  const sortableOptions = options.slice(fixedOptionsStart);
+  return (
+    <DndContainer
+      sortedOptions={sortableOptions}
+      getId={option => option.id}
+      onChange={sortedOptions => onChange([...staticOptions, ...sortedOptions])}
+      renderOption={props => (
+        <li
+          className={clsx(styles.option, props.isDragging && styles.placeholder, props.isSorting && styles.sorting)}
+          style={props.style}
+        >
+          {renderOption(props)}
+        </li>
+      )}
+      renderActiveOption={props => <Box>{renderOption(props)}</Box>}
+      i18nStrings={i18nStrings}
+    >
+      {content => (
+        <ul className={styles.list} aria-label="re-orderable list" role="list">
+          {staticOptions.map(option => (
+            <React.Fragment key={option.id}>{renderStaticOption?.(option)}</React.Fragment>
```

Comment:
Minor as this is a dev page: when the first 3 items are marked as fixed:
- Their visual height decreases by about 1px
- They become divs immediately nested in an unordered list (whereas technically only `li`, `script` or `template` are allowed: see [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul#technical_summary))

Reply:
Does `renderStaticOption` actually bring much? You can just customize the rendering in `renderOption`

Reply:
> Their visual height decreases by about 1px

This seems to be because of [this 1px transparent bottom border](https://github.com/cloudscape-design/components/pull/3091/files#diff-2f72ef25158ce8220ac390da383d63cfc322a7ba26a7efabef5c19a726786259R42) applied when sorting.

Reply:
Great observation! I will fix that by adding the li with style.option to make the rendering identical.

Yes, the renderOption can be customised but I want the first N items to be fixed in their positions. If I just disable the drag handles while rendering the options inside dnd container the user will be still able to change the order of those by moving other items around.

Reply:
OK, thanks for explaining. Still a very minor, non-blocking issue: while dragging a list item, the list becomes 1px shorter —the last item probably has a 1px transparent bottom border only while not dragging.

Reply:
Thanks, updated!

Reply:
Again, not blocking this because it's an internal dev page, but now the borders between list items have rounded corners :)